### PR TITLE
Svelte: fixup popover placement and root path rendering

### PR DESCRIPTION
**Pull request by camdencheek**
This offsets the file popover so the file names line up. Additionally, it hides the directory name when the file is located in the root directory because it looks awkward.

screenshot-2024-05-21_17-07-40@2x

# Test plan

Manual visual testing.

### CODE REVIEW

File: client/web-sveltekit/src/lib/Popover.svelte [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
     import { popover, onClickOutside, portal } from './dom'
 
-    export let placement: Placement = 'bottom'
     /**
      * Show the popover when hovering over the trigger.
      */
     export let showOnHover = false
+    export let placement: Placement = 'bottom'
+    export let offset: OffsetOptions = showOnHover ? 0 : 3
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/Popover.svelte:12]: This exposes the offset options, which can be used in conjunction with `Placement`. The default is the same default as before

File: client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/FileTree.svelte [hunk 2]
```diff
@@ -124,7 +124,15 @@
                     We handle navigation via the TreeView's select event, to preserve the focus state.
                     Using a link here allows us to benefit from data preloading.
                 -->
-                <Popover let:registerTrigger placement="right-end" showOnHover>
+                <Popover
+                    let:registerTrigger
+                    placement="right-start"
+                    offset={{
+                        // A number which visually offsets popover header so the file names line up
+                        crossAxis: -37,
```
Comment (camdencheek) [inline client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/FileTree.svelte:132]: This is a little awkward, but it doesn't need to be perfect so it's okay if it gets out of sync. I think an ideal solution here would probably look like "this point in my popover component defines the ideal anchor point"

**Review by taiyab — COMMENTED** [review]
Visually: LGTM!

**Comment by vovakulikov** [comment]
@taiyab @camdencheek  not a blocker for this PR but since we're changing file tooltip here, 5c of my feeback. 

When I saw it for the first time I thought that the commit badge was in the incorrect place, I would expect it right after the "Last changed @" line, like this 

Screenshot 2024-05-22 at 11 18 35

Also, I would try to add an arrow UI to the tooltip that points to the target (file entry in the file tree) similar to what we have in the tooltip (also not a blocker to this PR)

**Comment by camdencheek** [comment]
@taiyab any thoughts on an arrow for the popover component specifically? Our tooltips have an arrow, but it was never added for the popovers

**Comment by taiyab** [comment]
> When I saw it for the first time I thought that the commit badge was in the incorrect place, I would expect it right after the "Last changed @" line, like this

I'd rather try and keep this as consistent as possible to how we (ought to) render the commits in other places too (history list, etc). So, instead of adding the SHA on the "Last changed" line, we just remove the "@" which is causing the confusion.

> any thoughts on an arrow for the popover component specifically? Our tooltips have an arrow, but it was never added for the popovers

No need for this. Thanks.

**Comment by camdencheek** [comment]
Closing since we are removing file popovers from the file tree

**Pull request closed without merging**